airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -70,75 +71,75 @@ class SnowflakeDestinationHandler(
     private fun getFinalTableRowCount(
         streamIds: List<StreamId>
     ): LinkedHashMap<String, LinkedHashMap<String, Int>> {
-        val tableRowCounts = LinkedHashMap<String, LinkedHashMap<String, Int>>()
-        // convert list stream to array
-        val namespaces = streamIds.map { it.finalNamespace }.toTypedArray()
-        val names = streamIds.map { it.finalName }.toTypedArray()
-        val query =
-            """
-            |SELECT table_schema, table_name, row_count
-            |FROM information_schema.tables
-            |WHERE table_catalog = ? 
-            |AND table_schema IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-            |AND table_name IN (${IntRange(1, streamIds.size).joinToString { "?" }})
-            |""".trimMargin()
-        val bindValues = arrayOf(databaseName) + namespaces + names
-        val results: List<JsonNode> = database.queryJsons(query, *bindValues)
-        for (result in results) {
-            val tableSchema = result["TABLE_SCHEMA"].asText()
-            val tableName = result["TABLE_NAME"].asText()
-            val rowCount = result["ROW_COUNT"].asInt()
-            tableRowCounts
-                .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }[tableName] =
-                rowCount
+
+        try {
+            val tableRowCountsFromShowQuery = LinkedHashMap<String, LinkedHashMap<String, Int>>()
+
+            for (stream in streamIds) {
+                val showColumnsQuery =
+                    """
+                    SHOW TABLES LIKE '${stream.finalName}' IN "$databaseName"."${stream.finalNamespace}";
+                    """.trimIndent()
+                val showColumnsResult = database.queryJsons(
+                    showColumnsQuery,
+                )
+                for (result in showColumnsResult) {
+                    val tableSchema = result["schema_name"].asText()
+                    val tableName = result["name"].asText()
+                    val rowCount = result["rows"].asText()
+
+                    tableRowCountsFromShowQuery
+                        .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }[tableName] =
+                        rowCount.toInt()
+                }
+            }
+            return tableRowCountsFromShowQuery
+        } catch (e: SnowflakeSQLException) {
```

Comment:
this seems to be the wrong level to catch that exception, no? I'd expect we need to catch that exception for each individual stream, rather than catch-all and return 0 for every table